google/cloud/internal/rest_client_integration_test.cc

```diff
@@ -0,0 +1,336 @@
+  VerifyJsonPayloadResponse("PUT", json_payload_, std::move(response_status));
+}
+
+TEST_F(RestClientIntegrationTest, AnythingPutJsonContentTypeTwoSpans) {
+  options_.set<UnifiedCredentialsOption>(MakeInsecureCredentials());
+  auto client = DefaultRestClient::GetRestClient(url_, options_);
+  RestRequest request;
+  request.SetPath("anything");
+
+  std::string payload1 = json_payload_.substr(0, json_payload_.size() / 4);
+  // Naive attempt to avoid the data pointers of payload1 and payload2 being
+  // contiguous.
+  std::string gap = std::string(4096 * 4096, '/');
+  std::string payload2 =
+      json_payload_.substr(json_payload_.size() / 4, json_payload_.size());
+
+  absl::Span<char const> span1 = absl::MakeConstSpan(payload1);
+  absl::Span<char const> span2 = absl::MakeConstSpan(payload2);
+  request.AddHeader("content-type", "application/json");
+  auto response_status = client->Put(request, {span1, span2});
+  VerifyJsonPayloadResponse("PUT", json_payload_, std::move(response_status));
+}
+
+TEST_F(RestClientIntegrationTest, AnythingPutJsonContentTypeEmptyMiddleSpan) {
+  options_.set<UnifiedCredentialsOption>(MakeInsecureCredentials());
+  auto client = DefaultRestClient::GetRestClient(url_, options_);
+  RestRequest request;
+  request.SetPath("anything");
+
+  std::string payload1 = json_payload_.substr(0, json_payload_.size() / 4);
+  std::string empty;
+  std::string payload2 =
+      json_payload_.substr(json_payload_.size() / 4, json_payload_.size());
+
+  absl::Span<char const> span1 = absl::MakeConstSpan(payload1);
+  absl::Span<char const> empty_span = absl::MakeConstSpan(empty);
+  absl::Span<char const> span2 = absl::MakeConstSpan(payload2);
+  request.AddHeader("content-type", "application/json");
+  auto response_status = client->Put(request, {span1, empty_span, span2});
+  VerifyJsonPayloadResponse("PUT", json_payload_, std::move(response_status));
+}
+
+TEST_F(RestClientIntegrationTest, AnythingPutJsonContentTypeEmptyFirstSpan) {
+  options_.set<UnifiedCredentialsOption>(MakeInsecureCredentials());
+  auto client = DefaultRestClient::GetRestClient(url_, options_);
+  RestRequest request;
+  request.SetPath("anything");
+
+  std::string payload1 = json_payload_.substr(0, json_payload_.size() / 4);
+  std::string empty;
+  std::string payload2 =
+      json_payload_.substr(json_payload_.size() / 4, json_payload_.size());
+
+  absl::Span<char const> span1 = absl::MakeConstSpan(payload1);
+  absl::Span<char const> empty_span = absl::MakeConstSpan(empty);
+  absl::Span<char const> span2 = absl::MakeConstSpan(payload2);
+  request.AddHeader("content-type", "application/json");
+  auto response_status = client->Put(request, {empty_span, span1, span2});
+  VerifyJsonPayloadResponse("PUT", json_payload_, std::move(response_status));
+}
+
+TEST_F(RestClientIntegrationTest, ResponseBodyLargerThanSpillBuffer) {
+  std::string large_json_payload =
+      ReadFile("google/cloud/testing_util/rest_testing_large_json.json");
```

Comment:
For these purposes you can create a large JSON object using something like:

```cc
  nlohmann::json json;
  for (int i = 0; i != 1000; ++i) {
    json["row" + std::to_string(i)] = std::string('A', 128');
  }
 auto large_json_payload = json.dumps();
```

No need to commit a large test file?

It is not clear that your large payload needs to be a JSON object either.  You could just test with a large `content-type: application/octet-stream`.